[feature] GitHub Pages

obieler opened the pull request:
Close #17 

https://epfl-si.github.io/startpage

Harpogma committed:
[feature] WIP to host the site on GitHub Pages

obieler committed:
Fix background command

obieler committed:
Try github-pages-deploy-action

obieler committed:
Specify the folder

obieler committed:
Set main branch

ponsfrilus commented:
I though that [ejs](https://ejs.co/) had a build process that could be used to generate a static website. It seems that the ejs CLI got something (https://stackoverflow.com/a/30978686/960623), but this is not really convenient. In the end, we have to ask ourself whether ejs was a good choice or not; if the templating language has to be changed, the sooner the better.

williambelle commented on .github/workflows/githubpages.yml:
```diff
@@ -0,0 +1,40 @@
+name: Deploy GitHub Pages
```
:warning:

obieler replied:
Fixed :+1:

williambelle reviewed (commented):
We should remove (hide) the cookie consent banner.

williambelle commented:
> I though that [ejs](https://ejs.co/) had a build process that could be used to generate a static website. It seems that the ejs CLI got something ([stackoverflow.com/a/30978686/960623](https://stackoverflow.com/a/30978686/960623)), but this is not really convenient. In the end, we have to ask ourself whether ejs was a good choice or not; if the templating language has to be changed, the sooner the better.

In my opinion, the choice of template language has nothing to do with it.

We can generate the templates as follows:

```sh
./node_modules/.bin/ejs ./src/views/pages/home.ejs -o index.html
./node_modules/.bin/ejs ./src/views/pages/catalog.ejs -i "{\"data\": `cat src/data/EPFL-services.json`}" -o catalog.html
```

I see no point in maintaining a (complicated) build process to generate a static site in the medium-term. 
GitHub Pages should only be used for the short-term.

ponsfrilus committed:
[fix] invisible chars in `searchEngine` variable (#25)

The `searchEngine` used to select the action for each search wasn't
using a proper trim function, resulting in invalid match due to spaces.

ponsfrilus committed:
[feature] new search (#29)

* [feature] add "EPFL map" icon

* [feature] add "EPFL people" icon

* [feature] add "EPFL units" icon

* [feature] add "actu" search

* [feature] add EPNET search

ponsfrilus committed:
[feature] first changelog (#27)

ponsfrilus committed:
[feature] a minimalist footer (#26)

This add a footer with link to
- about page
- the version (linked to GitHub release)
- GitHub issue
- help
- source

Update the catalog page with new footer.

Closes #10

obieler committed:
Update workflow title

obieler committed:
Missing word

obieler commented:
> We should remove (hide) the cookie consent banner.

Yes, I will do it in a separate PR.
